_posts/2023-01-04-trino-2022-the-rabbit-reflects.md

```diff
@@ -0,0 +1,579 @@
+ various connectors. Support for setting comments on views was introduced very
+ recently and includes support for Hive and Iceberg. 
+* A ton of new functions were added, including `to_base32`, `from_base32`,
+ `trim_array`, and `trim`.
+
+### Performance improvements
+
+Despite all the hype about vectorization being a silver bullet to make databases
+go fast, the real speed comes from [better algorithms and better data structures
+that lead to lower resource consumption](https://www.youtube.com/clip/UgkxQwDYDS6evVJelNVjWAgrIhzg_Q-cAEyq).
+Following is a list of some improvements that made their way into Trino this
+year:
+
+* Trino now offers improved performance for a variety of operations, including
+ complex join criteria pushdown to connectors, faster aggregations, faster
+ joins, and better performance for large clusters. We have also implemented
+ improvements specifically for aggregations with filters and for the Glue
+ metastore. In addition, we now support dynamic filtering for various connectors
+ and have faster query planning for the Hive, Delta Lake, Iceberg, MySQL,
+ PostgreSQL, and SQL Server connectors.
+* Along with general performance optimizations, there have been a great deal of
+ query planning optimizations that lead to better performance for specific SQL
+ operators. These include faster `INSERT` queries, improved performance for
+ `LIKE` expressions and highly selective `LIMIT` queries, and enhanced
+ performance and reliability for `INSERT` and `MERGE` operations. We also made
+ performance improvements for `JOIN`, `UNION`, and `GROUP BY` queries, as well
+ as faster planning of queries with `IN` predicates. 
+* There are also optimizations for specific SQL types’ performance, such as
+ string, `DECIMAL`, `MAP` and `ROW` types. We have also made aggregations over 
+ `DECIMAL` columns faster and improved the performance of `ROW` type and
+ aggregation.
+* A last set of improvements come from reading open file formats like ORC and
+ Parquet efficiently. We have improved the speed of reading or writing of all 
+ data types from and to Parquet in general. There were also general performance
+ to ORC types, and now have the ability to write Bloom filters in ORC files. We
+ have also improved performance and efficiency for a wide range of ORC and
+ Parquet-related operations.
+
+These improvements in aggregate are at the core of what makes Trino fast. There
+is no silver bullet you can plug in to speed things up. It takes time, effort,
+and smart changes to improve the speed of various systems.
+
+### Runtime improvements
+
+[Trino upgraded Java versions](https://github.com/trinodb/trino/issues/9876)from
+Java 11 to the next long-term support version of Java, which was Java 17. This
+upgrade improves the latency, memory footprint, startup, ramp up, pause times,
+and shut down times for Trino. Along with these performance enhancements one of the other
```

Comment:
The rest is too long .. essentially it just needs to say that the GC in Java 17 is also improved